File: packages/jsii-pacmak/test/expected.jsii-calc/java/src/main/java/software/amazon/jsii/tests/calculator/ExtendsInternalInterface.java
package software.amazon.jsii.tests.calculator;

/**
 * EXPERIMENTAL
 */
@javax.annotation.Generated(value = "jsii-pacmak")
@software.amazon.jsii.Jsii(module = software.amazon.jsii.tests.calculator.$Module.class, fqn = "jsii-calc.ExtendsInternalInterface")
@software.amazon.jsii.Jsii.Proxy(ExtendsInternalInterface.Jsii$Proxy.class)
@software.amazon.jsii.Stability(software.amazon.jsii.Stability.Level.Experimental)
public interface ExtendsInternalInterface extends software.amazon.jsii.JsiiSerializable {

    /**
     * EXPERIMENTAL
     */
    @software.amazon.jsii.Stability(software.amazon.jsii.Stability.Level.Experimental)
    @org.jetbrains.annotations.NotNull java.lang.Boolean getBoom();

    /**
     * EXPERIMENTAL
     */
    @software.amazon.jsii.Stability(software.amazon.jsii.Stability.Level.Experimental)
    @org.jetbrains.annotations.NotNull java.lang.String getProp();

    /**
     * @return a {@link Builder} of {@link ExtendsInternalInterface}
     */
    @software.amazon.jsii.Stability(software.amazon.jsii.Stability.Level.Experimental)
    static Builder builder() {
        return new Builder();
    }
    /**
     * A builder for {@link ExtendsInternalInterface}
     */
    @software.amazon.jsii.Stability(software.amazon.jsii.Stability.Level.Experimental)
    public static final class Builder implements software.amazon.jsii.Builder<ExtendsInternalInterface> {
        private java.lang.Boolean boom;
        private java.lang.String prop;

        /**
         * Sets the value of {@link ExtendsInternalInterface#getBoom}
         * @param boom the value to be set. This parameter is required.
         * @return {@code this}
         */
        @software.amazon.jsii.Stability(software.amazon.jsii.Stability.Level.Experimental)
        public Builder boom(java.lang.Boolean boom) {
            this.boom = boom;
            return this;
        }

        /**
         * Sets the value of {@link ExtendsInternalInterface#getProp}
         * @param prop the value to be set. This parameter is required.
         * @return {@code this}
         */
        @software.amazon.jsii.Stability(software.amazon.jsii.Stability.Level.Experimental)
        public Builder prop(java.lang.String prop) {
            this.prop = prop;
            return this;
        }

        /**
         * Builds the configured instance.
         * @return a new instance of {@link ExtendsInternalInterface}
         * @throws NullPointerException if any required attribute was not provided
         */
        @software.amazon.jsii.Stability(software.amazon.jsii.Stability.Level.Experimental)
        @Override
        public ExtendsInternalInterface build() {
            return new Jsii$Proxy(boom, prop);
        }
    }

    /**
     * An implementation for {@link ExtendsInternalInterface}
     */
    @software.amazon.jsii.Stability(software.amazon.jsii.Stability.Level.Experimental)
    final class Jsii$Proxy extends software.amazon.jsii.JsiiObject implements ExtendsInternalInterface {
        private final java.lang.Boolean boom;
        private final java.lang.String prop;

        /**
         * Constructor that initializes the object based on values retrieved from the JsiiObject.
         * @param objRef Reference to the JSII managed object.
         */
        protected Jsii$Proxy(final software.amazon.jsii.JsiiObjectRef objRef) {
            super(objRef);
            this.boom = this.jsiiGet("boom", java.lang.Boolean.class);
            this.prop = this.jsiiGet("prop", java.lang.String.class);
        }

        /**
         * Constructor that initializes the object based on literal property values passed by the {@link Builder}.
         */
        private Jsii$Proxy(final java.lang.Boolean boom, final java.lang.String prop) {
            super(software.amazon.jsii.JsiiObject.InitializationMode.JSII);
            this.boom = java.util.Objects.requireNonNull(boom, "boom is required");
            this.prop = java.util.Objects.requireNonNull(prop, "prop is required");
        }

        @Override
        public java.lang.Boolean getBoom() {
            return this.boom;
        }

        @Override
        public java.lang.String getProp() {
            return this.prop;
        }

        @Override
        public com.fasterxml.jackson.databind.JsonNode $jsii$toJson() {
            final com.fasterxml.jackson.databind.ObjectMapper om = software.amazon.jsii.JsiiObjectMapper.INSTANCE;
            final com.fasterxml.jackson.databind.node.ObjectNode data = com.fasterxml.jackson.databind.node.JsonNodeFactory.instance.objectNode();

            data.set("boom", om.valueToTree(this.getBoom()));
            data.set("prop", om.valueToTree(this.getProp()));

            final com.fasterxml.jackson.databind.node.ObjectNode struct = com.fasterxml.jackson.databind.node.JsonNodeFactory.instance.objectNode();
            struct.set("fqn", om.valueToTree("jsii-calc.ExtendsInternalInterface"));
            struct.set("data", data);

            final com.fasterxml.jackson.databind.node.ObjectNode obj = com.fasterxml.jackson.databind.node.JsonNodeFactory.instance.objectNode();
            obj.set("$jsii.struct", struct);

            return obj;
        }

        @Override
        public boolean equals(Object o) {
            if (this == o) return true;
            if (o == null || getClass() != o.getClass()) return false;

            ExtendsInternalInterface.Jsii$Proxy that = (ExtendsInternalInterface.Jsii$Proxy) o;

            if (!boom.equals(that.boom)) return false;
            return this.prop.equals(that.prop);
        }

        @Override
        public int hashCode() {
            int result = this.boom.hashCode();
            result = 31 * result + (this.prop.hashCode());
            return result;
        }
    }
}

File: packages/jsii-pacmak/test/expected.jsii-calc/kotlin/src/main/kotlin/software/amazon/jsii/tests/kotlin/calculator/ExtendsInternalInterface.kt
package software.amazon.jsii.tests.kotlin.calculator

@javax.annotation.Generated("jsii-pacmak")
@software.amazon.jsii.Stability(software.amazon.jsii.Stability.Level.Experimental)
interface ExtendsInternalInterface : software.amazon.jsii.JsiiSerializable {
    @get:software.amazon.jsii.Stability(software.amazon.jsii.Stability.Level.Experimental)
    val boom: kotlin.Boolean

    @get:software.amazon.jsii.Stability(software.amazon.jsii.Stability.Level.Experimental)
    val prop: kotlin.String

    class Builder {
        @get:software.amazon.jsii.Stability(software.amazon.jsii.Stability.Level.Experimental)
        var boom: kotlin.Boolean? = null

        @get:software.amazon.jsii.Stability(software.amazon.jsii.Stability.Level.Experimental)
        var prop: kotlin.String? = null

        fun build(): software.amazon.jsii.tests.kotlin.calculator.ExtendsInternalInterface {
            val boom = this.boom ?: kotlin.error("Value for property 'boom' must be specified")
            val prop = this.prop ?: kotlin.error("Value for property 'prop' must be specified")
            return `Jsii$Proxy`(boom, prop)
        }
    }

    class `Jsii$Proxy` : software.amazon.jsii.JsiiObject, software.amazon.jsii.tests.kotlin.calculator.ExtendsInternalInterface {
        @get:software.amazon.jsii.Stability(software.amazon.jsii.Stability.Level.Experimental)
        override val boom: kotlin.Boolean

        @get:software.amazon.jsii.Stability(software.amazon.jsii.Stability.Level.Experimental)
        override val prop: kotlin.String

        protected constructor(objRef: software.amazon.jsii.JsiiObjectRef) : super(objRef) {
            this.boom = this.jsiiCall("boom", kotlin.Boolean::class.java) ?: error("'boom' should be present")
            this.prop = this.jsiiCall("prop", kotlin.String::class.java) ?: error("'prop' should be present")
        }

        constructor(boom: kotlin.Boolean, prop: kotlin.String) : super(software.amazon.jsii.JsiiObject.InitializationMode.JSII) {
            this.boom = boom
            this.prop = prop
        }

        override fun equals(other: Any?): Boolean {
            if (this === other) return true
            if (javaClass != other?.javaClass) return false
            other as software.amazon.jsii.tests.kotlin.calculator.ExtendsInternalInterface
            if (boom != other.boom) return false
            if (prop != other.prop) return false
            return true
        }

        override fun hashCode(): Int {
            var result = this.boom.hashCode()
            result = 31 * result + (this.prop.hashCode())
            return result
        }

        override fun `$jsii$toJson`(): com.fasterxml.jackson.databind.JsonNode {
            val om = software.amazon.jsii.JsiiObjectMapper.INSTANCE
            val obj = com.fasterxml.jackson.databind.node.JsonNodeFactory.instance.objectNode()
            obj.set<com.fasterxml.jackson.databind.JsonNode>("boom", om.valueToTree(this.boom))
            obj.set<com.fasterxml.jackson.databind.JsonNode>("prop", om.valueToTree(this.prop))
            return obj
        }
    }
}
